modules/tinymce/src/plugins/table/test/ts/browser/command/MergeCellCommandTest.ts

```diff
@@ -35,68 +31,153 @@ UnitTest.asynctest('browser.tinymce.plugins.table.command.MergeCellCommandTest',
           '<tr><td data-mce-selected="1">a2</td><td data-mce-selected="1" data-mce-last-selected="1">b2</td></tr>' +
           '<tr><td>a3</td><td>b3</td></tr>' +
           '</tbody>' +
           '</table>'
-        ),
-        after: (
-          '<table>' +
+      ),
+      after: (
+        '<table>' +
           '<tbody>' +
           '<tr><td colspan="2" rowspan="2">a1<br />b1<br />a2<br />b2</td></tr>' +
           '<tr></tr>' +
           '<tr><td>a3</td><td>b3</td></tr>' +
           '</tbody>' +
+        '</table>'
+      ),
+      expectedEvents: [ defaultEvent ],
+    });
+  });
+
+  it('TBA: Should merge b3+c3 but not reduce a2a3', () => {
+    const editor = hook.editor();
+    testMerge(editor, {
+      before: (
+        '<table>' +
+            '<tbody>' +
+              '<tr>' +
+                '<td>a1</td>' +
+                '<td>b1</td>' +
+                '<td>c1</td>' +
+              '</tr>' +
+              '<tr>' +
+                '<td rowspan="2">a2a3</td>' +
+                '<td>b2</td>' +
+                '<td>c2</td>' +
+              '</tr>' +
+              '<tr>' +
+                '<td data-mce-selected="1" data-mce-first-selected="1">b3</td>' +
+                '<td data-mce-selected="1" data-mce-last-selected="1">c3</td>' +
+              '</tr>' +
+            '</tbody>' +
           '</table>'
-        ),
-        expectedEvents: [ defaultEvent ],
-      },
-      /*
+      ),
+      after: (
+        '<table>' +
+          '<tbody>' +
+            '<tr>' +
+              '<td>a1</td>' +
+              '<td>b1</td>' +
+            '<td>c1</td>' +
+            '</tr>' +
+            '<tr>' +
+              '<td rowspan="2">a2a3</td>' +
+              '<td>b2</td>' +
+              '<td>c2</td>' +
+            '</tr>' +
+            '<tr>' +
+              '<td colspan="2">b3<br />c3</td>' +
+            '</tr>' +
+          '</tbody>' +
+        '</table>'
+      ),
+      expectedEvents: [ defaultEvent ],
+    });
+  });
+
+  it('TINY-6901: Will apply correct size to cell with colspan after cell merge', () => {
+    const editor = hook.editor();
+    const before = (
+      '<table style="width: 25.4582%;">' +
+        '<tbody>' +
+          '<tr>' +
+            '<td style="width: 42.7414%;" data-mce-selected="1" data-mce-first-selected="1"></td>' +
+            '<td style="width: 51.8049%;" data-mce-selected="1" data-mce-last-selected="1"></td>' +
+          '</tr>' +
+          '<tr>' +
+            '<td style="width: 42.7414%;"></td>' +
+            '<td style="width: 51.8049%;"></td>' +
+          '</tr>' +
+        '</tbody>' +
+      '</table>'
+    );
+
+    editor.setContent(before);
+    const cols = editor.dom.select('td[data-mce-selected]');
+    const totalColsWidth = Arr.foldl(cols, (acc, col) => acc + getWidthNumber(Css.getRaw(SugarElement.fromDom(col), 'width').getOrDie()), 0);
```

Comment:
```suggestion
    const cols = SelectorFilter.descendant(TinyDom.body(editor), 'td[data-mce-selected]');
    const totalColsWidth = Arr.foldl(cols, (acc, col) => acc + getWidthNumber(Css.getRaw(col, 'width').getOrDie()), 0);
```
This just saves having to convert and we normally prefer sugar functions where possible.